googlesource/googlesource.go

```diff
@@ -334,79 +280,40 @@ func (s *Source) Next(ctx context.Context) (sdk.Record, error) {
-	s.positions.lock.Lock()
+	s.position = ""
 
-	err := json.Unmarshal(pos, &s.positions.positions)
+	err := json.Unmarshal(pos, &s.position)
 	if err != nil {
 		sdk.Logger(s.ctx).Info().Msg("Could not get position. Will start with offset 0")
 	}
-	s.positions.lock.Unlock()
-}
-
-func getTables(s *Source) (err error) {
-	if s.sourceConfig.Config.TableIDs == "" {
-		s.tables, err = s.listTables(s.sourceConfig.Config.ProjectID, s.sourceConfig.Config.DatasetID)
-		if err != nil {
-			sdk.Logger(s.ctx).Trace().Str("err", err.Error()).Msg("error found while listing table")
-		}
-	} else {
-		s.tables = strings.Split(s.sourceConfig.Config.TableIDs, ",")
-	}
-	return err
 }
 
 func (s *Source) runIterator() (err error) {
-	rowInput := make(chan readRowInput)
-
-	if err = getTables(s); err != nil {
-		sdk.Logger(s.ctx).Trace().Str("err", err.Error()).Msg("error found while fetching tables. Need to stop proccessing ")
+	// Snapshot sync. Start were we left last
+	ctx := s.ctx
+	err = s.ReadGoogleRow(ctx)
+	if err != nil {
+		sdk.Logger(ctx).Trace().Str("err", err.Error()).Msg("error found while reading google row.")
 		return err
 	}
 
-	var wg sync.WaitGroup
-	// Snapshot sync. Start were we left last
-	for _, tableID := range s.tables {
-		wg.Add(1)
-
-		s.tomb.Go(func() (err error) {
-			sdk.Logger(s.ctx).Trace().Msg(fmt.Sprintf("position %v : %v", tableID, s.getPosition()[tableID]))
-			return s.ReadGoogleRow(rowInput, s.records)
-		})
-		position := s.getPosition()
-		rowInput <- readRowInput{tableID: tableID, offset: position[tableID], positions: position, wg: &wg}
-	}
-	wg.Wait()
 	for {
 		select {
 		case <-s.tomb.Dying():
 			return s.tomb.Err()
 		case <-s.ticker.C:
-			sdk.Logger(s.ctx).Trace().Msg("ticker started ")
-
-			if err = getTables(s); err != nil {
-				sdk.Logger(s.ctx).Trace().Str("err", err.Error()).Msg("error found while fetching tables. Need to stop proccessing ")
-				return err
+			sdk.Logger(ctx).Trace().Msg("ticker started ")
+			err = runCDCIterator(ctx, s)
+			if err != nil {
+				sdk.Logger(ctx).Trace().Msg(fmt.Sprintf("error found %v", err))
+				return
 			}
-			runCDCIterator(s, rowInput)
 		}
 	}
 }
 
-func runCDCIterator(s *Source, rowInput chan readRowInput) {
-	// wait group make sure that we start new iteration only
-	//  after the first iteration is completely done.
-	var wg sync.WaitGroup
-	for _, tableID := range s.tables {
-		wg.Add(1)
-		offset := s.getPosition()[tableID]
-		position := s.getPosition()
-		s.tomb.Go(func() (err error) {
-			sdk.Logger(s.ctx).Trace().Msg(fmt.Sprintf("position %v : %v", tableID, s.getPosition()[tableID]))
-			return s.ReadGoogleRow(rowInput, s.records)
-		})
-		rowInput <- readRowInput{tableID: tableID, offset: offset, positions: position, wg: &wg}
-	}
-	wg.Wait()
+func runCDCIterator(ctx context.Context, s *Source) (err error) {
+	sdk.Logger(ctx).Trace().Msg(fmt.Sprintf("position %v : %v", s.sourceConfig.Config.TableID, s.position))
+	err = s.ReadGoogleRow(ctx)
+	return
```

Comment:
All this method does is to call another method with the same args, so it looks like this method is redundant.